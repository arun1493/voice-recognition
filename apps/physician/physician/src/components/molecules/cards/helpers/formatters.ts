import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
