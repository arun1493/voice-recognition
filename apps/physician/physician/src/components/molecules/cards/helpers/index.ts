export * from "./labels";
export * from "./formatters";
