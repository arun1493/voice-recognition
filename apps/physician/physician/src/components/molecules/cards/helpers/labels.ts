import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
