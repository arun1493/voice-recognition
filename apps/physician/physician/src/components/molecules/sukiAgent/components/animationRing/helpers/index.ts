export * from "./ringSize";
export * from "./animation";
