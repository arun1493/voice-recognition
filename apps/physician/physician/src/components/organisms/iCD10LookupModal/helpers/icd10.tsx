import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
