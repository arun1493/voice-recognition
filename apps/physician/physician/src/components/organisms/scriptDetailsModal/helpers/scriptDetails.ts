import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
