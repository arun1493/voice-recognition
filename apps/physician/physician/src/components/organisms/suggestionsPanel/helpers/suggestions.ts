import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
