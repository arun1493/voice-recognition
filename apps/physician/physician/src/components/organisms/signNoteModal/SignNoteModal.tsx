import store from "store2";
import _includes from "lodash/includes";
import { useTranslation } from "react-i18next";
import { useLocation, useNavigate } from "react-router-dom";
import { batch, useDispatch, useSelector } from "react-redux";
import { ClientEmitEventTypes } from "@suki-web/suki-business";
import { FunctionComponent, useEffect, useState } from "react";
import { Dialog, DialogButtonsProps, Icon, Text } from "@suki-web/suki-ui";

import {
  ConfirmICDCodes,
  InitialOrSubmitModal,
  MissingICDCodes,
  MultipleEncounters,
  SubmissionWarning,
  UnableToReachEmr
} from "./components";
import {
  appointmentReader,
  compositionReader,
  fhirTokenCacheReader,
  metadataReader,
  organizationReader,
  patientReader,
  userReader
} from "../../../readers";
import {
  getNoEncounterModalDetails,
  getRoute,
  isEnhancedReviewEnabled,
  isNoteHasDictation,
  isSecondaryEmr
} from "../../../helpers";
import {
  Actor,
  DESTINATIONS_TO_SEND_NOTE,
  EmrDestination,
  LOCAL_STORE,
  Pages,
  QUERY_NAMES,
  SecondaryEmr,
  SUBMISSION_WARNING,
  SubmissionModals,
  TOAST_TYPES,
  USER_REQUEST_TYPE
} from "../../../constants";
import { queryClient } from "../../../app";
import { useSocket } from "../../../socket";
import { toastsActions } from "../../../redux/toast";
import { selectEmrTokens } from "../../../redux/emr";
import { selectUserDetails } from "../../../redux/user";
import { getOrganizationState } from "../../../redux/organization";
import { getModalsState, modalsActions } from "../../../redux/modals";
import { noteActions, selectCurrentComposition } from "../../../redux/note";
import { handleNavigateBackFromNote as navigateAfterSubmission } from "../../../helpers";

const SignNoteModal: FunctionComponent = () => {
  const navigate = useNavigate();
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const location = useLocation();
  const { transmitEvent } = useSocket();

  const { signNote } = useSelector(getModalsState);

  const composition = useSelector(selectCurrentComposition);
  const metadata = compositionReader.metadata(composition);

  const isInitial = signNote.type === SubmissionModals.INITIAL;
  const [isSubmittingInProgress, setIsSubmittingInProgress] = useState<boolean>(false);
  const [isSendingToChartInProgress, setIsSendingToChartInProgress] = useState<boolean>(false);

  const isPatientSelected = Boolean(metadataReader.patient(metadata));
  const emrType = metadataReader.emrType(metadata);

  const isEmr = isSecondaryEmr(emrType);
  const emrDestinations = metadataReader.emrDestinations(metadata);
  const [primaryDestination, secondaryDestination] = emrDestinations;

  const isAthena = emrType === SecondaryEmr.ATHENA_EMR;
  const isEpic = emrType === SecondaryEmr.EPIC_EMR;
  const isElation = emrType === SecondaryEmr.ELATION_EMR;

  const hasDictation = isNoteHasDictation(composition);

  const user = useSelector(selectUserDetails);
  const userEnhancedReview = userReader.enhancedReview(user);
  const organization = useSelector(getOrganizationState);
  const orgEnhancedReview = organizationReader.enhancedReview(organization);

  // check primary destination - because did note submittet to secondary we know only after got response from ws
  const isSendNoteToEmrAvailable = _includes(DESTINATIONS_TO_SEND_NOTE, primaryDestination);

  const isNoteEligibleForEnhancedReview =
    isEnhancedReviewEnabled(orgEnhancedReview, userEnhancedReview) && isInitial && hasDictation;

  const emrTokens = useSelector(selectEmrTokens);

  const closeModal = () => {
    dispatch(modalsActions.hideSignNoteModal());
    setIsSubmittingInProgress(false);
    setIsSendingToChartInProgress(false);
  };

  const showSubmitSuccessFlow = () => {
    const patient = metadataReader.patient(metadata);
    const message = t("modals.signNoteModal.noteEnhancedReviewNoOpenEncounter.text1");

    batch(() => {
      dispatch(toastsActions.add({ message, type: TOAST_TYPES.CONFIRMATION }));
      dispatch(noteActions.setCompositionStatus({ noteSubmissionSuccess: true }));
      dispatch(modalsActions.hideSignNoteModal());
    });

    queryClient.invalidateQueries([QUERY_NAMES.UNFINISHED_NOTES]);
    navigateAfterSubmission(location, navigate, patient);
  };

  const showConfirmCodesModal = () => {
    dispatch(modalsActions.showSignNoteModal({ type: SubmissionModals.CONFIRM_CODES }));
  };

  const showMissingCodesModal = () => {
    dispatch(modalsActions.showSignNoteModal({ type: SubmissionModals.MISSING_CODES }));
  };

  const setInitialStep = () => {
    dispatch(modalsActions.showSignNoteModal({ type: SubmissionModals.INITIAL }));
  };

  const setSubmitStep = () => {
    dispatch(modalsActions.showSignNoteModal({ type: SubmissionModals.SUBMIT }));
  };

  const submitNote = () => {
    if (isSendNoteToEmrAvailable || signNote.type === SubmissionModals.SUBMIT) {
      submitCompositionRequest();
      setIsSubmittingInProgress(true);
      return;
    }

    if (isInitial) {
      setSubmitStep();
    }
  };

  const submitCompositionRequest = (isEnhancedReview = false, isEmrChart = false) => {
    const noteId = compositionReader.compositionId(composition);
    const organizationId = compositionReader.organizationId(composition);
    const emrAccessToken = fhirTokenCacheReader.fhirAccessToken(emrTokens);

    const [primaryDestination, secondaryDestination] = emrDestinations;
    const tryAllDestinations = !(
      primaryDestination !== EmrDestination.DOCUMENT && secondaryDestination === EmrDestination.DOCUMENT
    );

    if (!isEnhancedReview && !isEmrChart) {
      let title;
      if (isAthena || isEpic) {
        title =
          primaryDestination !== EmrDestination.DOCUMENT
            ? t("modals.signNoteModal.saveNote.oneEncounterTip")
            : t("modals.signNoteModal.saveNote.sendingToEMRChartTip");
      } else if (isEmr) {
        title = t("modals.signNoteModal.saveNote.oneEncounterWithAllowSigning");
      }
      title && dispatch(modalsActions.setSignNoteModalSuccessTipMessage(title));
    }

    transmitEvent({
      type: ClientEmitEventTypes.SUBMIT_COMPOSITION_REQUEST,
      data: {
        actor: Actor.USER,
        bypassQa: !isEnhancedReview,
        emrAccessToken,
        isSignedOff: false,
        noteId,
        organizationId,
        requestTime: new Date().toISOString(),
        tryAllDestinations: isEmrChart ? isEmrChart : tryAllDestinations
      }
    });
  };

  const saveToEnhancedReview = () => {
    submitCompositionRequest(true);
    // TODO: [Sai]: Correct this flow once the dictation inside note is completed.
    // Please refer to https://sukiai.atlassian.net/browse/MOBAPP-437 for more details
    let title = t("modals.signNoteModal.saveNote.tipEnhancedReview");

    if (isAthena || isElation) {
      title = t("modals.signNoteModal.saveNote.tipEnhancedReviewForEmr");
    }

    dispatch(modalsActions.setSignNoteModalSuccessTipMessage(title));
    dispatch(modalsActions.setSignNoteModalIsEnhancedReview(true));
  };

  const saveUnfinishedNote = () => {
    const compositionId = compositionReader.compositionId(composition);

    const appointment = metadataReader.appointment(metadata);
    const patient = metadataReader.patient(metadata);
    const noteTypeId = metadataReader.noteTypeId(metadata);

    const patientId = patientReader.id(patient);
    const appointmentId = appointmentReader.id(appointment);

    transmitEvent({
      type: ClientEmitEventTypes.USER_REQUEST,
      data: {
        appointmentId,
        compositionId,
        noteTypeId,
        patientId,
        requestType: USER_REQUEST_TYPE.SAVE_COMPOSITION_TO_UNFINISHED
      }
    });

    const message = t("modals.signNoteModal.saveNote.saveToUnfinishedNotesTip");
    batch(() => {
      dispatch(toastsActions.add({ message, type: TOAST_TYPES.CONFIRMATION }));
      closeModal();
    });

    if (patientId) {
      const patientProfileRoute = getRoute(Pages.PatientProfile, { patientId });
      patientProfileRoute && navigate(patientProfileRoute);
    } else {
      navigate(-1);
    }
  };

  const sendToEMR = () => {
    submitCompositionRequest();
    const title = t("modals.signNoteModal.saveNote.oneEncounterTip");
    dispatch(modalsActions.setSignNoteModalSuccessTipMessage(title));
  };

  const sendToEMRChart = () => {
    submitCompositionRequest(false, true);
    const title = t("modals.signNoteModal.saveNote.sendingToEMRChartTip");
    dispatch(modalsActions.setSignNoteModalSuccessTipMessage(title));
    setIsSendingToChartInProgress(true);
  };

  const setInitialStepAndIgnoreMissingICDCodes = () => {
    batch(() => {
      dispatch(modalsActions.setSignNoteIgnoreMissingICDCodes());
      setInitialStep();
    });
  };

  const noPatientButtonProps = {
    end: { onClick: saveUnfinishedNote, text: t("modals.signNoteModal.noPatient.saveButton") },
    start: { onClick: closeModal, text: t("modals.signNoteModal.noPatient.cancelButton") }
  };

  const encounterButtonProps: DialogButtonsProps = {
    end: { text: t("modals.signNoteModal.noEncounter.saveToUnfinishedButton"), onClick: saveUnfinishedNote },
    hasCloseBtn: true
  };

  if (signNote.type === SubmissionModals.NO_ENCOUNTER_EMRCHART) {
    encounterButtonProps.start = {
      text: isSendingToChartInProgress ? (
        <>
          <Icon icon="loader" size="md" />
          {t("modals.signNoteModal.noEncounterEmrChart.submitting")}
        </>
      ) : (
        t("modals.signNoteModal.noEncounterEmrChart.submitToChart")
      ),
      onClick: sendToEMRChart
    };
  }

  const noteEnhancedReviewNoOpenEncounterButtonProps: DialogButtonsProps = {
    hasCloseBtn: false,
    end: {
      text: t("modals.signNoteModal.noteEnhancedReviewNoOpenEncounter.button.ok"),
      onClick: showSubmitSuccessFlow
    },
    start: {
      text: t("modals.signNoteModal.noteEnhancedReviewNoOpenEncounter.button.dontShow"),
      onClick: () => {
        store.set(LOCAL_STORE.HAS_DONT_SHOW_NO_OPEN_ENCOUNTER_AGAIN, true.toString());
        showSubmitSuccessFlow();
      }
    }
  };

  const { title, text1, text2 } = getNoEncounterModalDetails(signNote.type);

  useEffect(() => {
    return () => {
      setIsSubmittingInProgress(false);
      setIsSendingToChartInProgress(false);
    };
  }, []);

  if (!isPatientSelected) {
    return (
      <Dialog
        buttonProps={noPatientButtonProps}
        isOpen={!isPatientSelected}
        onClose={closeModal}
        title={t("modals.signNoteModal.noPatient.title")}
      >
        <Text size="md">{t("modals.signNoteModal.noPatient.text1")}</Text>
        <Text size="md">{t("modals.signNoteModal.noPatient.text2")}</Text>
      </Dialog>
    );
  }

  switch (signNote.type) {
    case SubmissionModals.NO_LAB_RESULTS:
      return (
        <SubmissionWarning
          closeModal={closeModal}
          saveUnfinishedNote={saveUnfinishedNote}
          warningType={SUBMISSION_WARNING.NO_LAB_RESULTS}
        />
      );

    case SubmissionModals.NO_VALID_OB_EPISODE:
      return (
        <SubmissionWarning
          closeModal={closeModal}
          saveUnfinishedNote={saveUnfinishedNote}
          warningType={SUBMISSION_WARNING.NO_VALID_OB_EPISODE}
        />
      );

    case SubmissionModals.CONFIRM_CODES:
      return (
        <ConfirmICDCodes
          closeModal={closeModal}
          setInitialStep={setInitialStep}
          showMissingCodesModal={showMissingCodesModal}
        />
      );

    case SubmissionModals.MISSING_CODES:
      return (
        <MissingICDCodes
          closeModal={closeModal}
          setInitialStepAndIgnoreMissingICDCodes={setInitialStepAndIgnoreMissingICDCodes}
          showConfirmCodesModal={showConfirmCodesModal}
        />
      );

    case SubmissionModals.INITIAL:
    case SubmissionModals.SUBMIT:
      return (
        <InitialOrSubmitModal
          closeModal={closeModal}
          hasEnhancedReview={isNoteEligibleForEnhancedReview}
          isInitial={isInitial}
          isSendNoteToEmrAvailable={isSendNoteToEmrAvailable}
          isSubmittingInProgress={isSubmittingInProgress}
          saveToEnhancedReview={saveToEnhancedReview}
          submitNote={submitNote}
        />
      );

    case SubmissionModals.NOTE_TO_ENHANCED_REVIEW:
      return (
        <Dialog
          buttonProps={noteEnhancedReviewNoOpenEncounterButtonProps}
          isOpen={true}
          onClose={closeModal}
          title={t("modals.signNoteModal.noteEnhancedReviewNoOpenEncounter.title")}
        >
          <Text size="md">{t("modals.signNoteModal.noteEnhancedReviewNoOpenEncounter.text1")}</Text>
          <Text size="md">{t("modals.signNoteModal.noteEnhancedReviewNoOpenEncounter.text2")}</Text>
        </Dialog>
      );

    case SubmissionModals.NO_ENCOUNTER_EMRCHART:
    case SubmissionModals.NO_ENCOUNTER:
      return (
        <Dialog
          buttonProps={encounterButtonProps}
          isLongFooter={signNote.type === SubmissionModals.NO_ENCOUNTER_EMRCHART}
          isOpen={true}
          onClose={closeModal}
          title={title}
        >
          <Text size="md">{text1}</Text>
          <Text size="md">{text2}</Text>
        </Dialog>
      );

    case SubmissionModals.MULTIPLE_ENCOUNTERS:
      return (
        <MultipleEncounters
          closeModal={closeModal}
          saveUnfinishedNote={saveUnfinishedNote}
          sendToEMR={sendToEMR}
          sendToEMRChart={sendToEMRChart}
        />
      );

    case SubmissionModals.UNABLE_TO_REACH_EMR:
    default:
      return <UnableToReachEmr closeModal={closeModal} />;
  }
};

export default SignNoteModal;

export { SignNoteModal };
