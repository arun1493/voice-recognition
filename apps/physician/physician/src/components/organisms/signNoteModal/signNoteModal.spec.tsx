import ReactDOM from "react-dom";
import { ReactPortal } from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";

import { SignNoteModal } from "./SignNoteModal";
import { configureI18n } from "../../../helpers";
import { componentSetupWithStore } from "../../../hocs";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";

const setup = componentSetupWithStore(() => <SignNoteModal />);

describe("SignNoteModal modal", () => {
  let baseElement: Element;
  let element: Element | null;

  beforeEach(() => {
    configureI18n();
    baseElement = setup().baseElement;
    ReactDOM.createPortal = jest.fn<ReactPortal, any>((element, node) => element);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    expect(baseElement).toBeTruthy();
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(SignNoteModal);
  });
});
