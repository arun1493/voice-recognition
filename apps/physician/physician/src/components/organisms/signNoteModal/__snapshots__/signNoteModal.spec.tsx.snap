// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignNoteModal modal should match snapshot 1`] = `
<div>
  <div
    className="modalMaskstyled__StyledBackdrop-sc-169llx0-0 fRODrN"
  />
  <dialog
    className="dialogstyled__StyledDialog-a4znt9-0 czGhrd"
  >
    <header
      className="dialogstyled__DialogHeader-a4znt9-1 ebNSod"
    >
      <h5
        className="headerstyled__StyledH5-sc-1y4xvz5-4 iITAex"
      >
        No Patient Name
      </h5>
    </header>
    <section
      className="dialogstyled__DialogContent-a4znt9-3 ffmWsl"
    >
      <p
        className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
        color="inherit"
        cursor="default"
        onClick={[Function]}
        size="md"
      >
        Patient name is not set, so note can't be sent to your EMR yet.
      </p>
      <p
        className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
        color="inherit"
        cursor="default"
        onClick={[Function]}
        size="md"
      >
        Save to "Unfinished Notes" to finish later?
      </p>
    </section>
    <footer
      className="dialogstyled__DialogFooter-a4znt9-4 jUbXLl"
    >
      <button
        className="dialogButtonstyled__StyledDialogButton-uvbi7e-0 kFNoNK"
        data-cy="dialog-button"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="dialogButtonstyled__StyledDialogButton-uvbi7e-0 kFNoNK"
        data-cy="dialog-button"
        onClick={[Function]}
      >
        Finish Later
      </button>
    </footer>
  </dialog>
</div>
`;
